### PR TITLE
Disable Identity Execution features for the 8.6 release

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

User creation, authorization patching and default user creation are all part of the 8.7 release. We should disabled them in 8.6 so we can guarantee our state is fresh for the 8.7 release.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Review by tmetzke — COMMENTED** [review]
👍 Changes look good to me.
Can we add a test case for each controller to verify what the response looks like in 8.6?

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/user/UserProcessors.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public static void addUserProcessors(
         .onCommand(
             ValueType.USER,
             UserIntent.UPDATE,
-            new UserUpdateProcessor(keyGenerator, processingState, writers, distributionBehavior))
-        .withListener(new DefaultUserCreator(processingState, config));
+            new UserUpdateProcessor(keyGenerator, processingState, writers, distributionBehavior));
+    //        .withListener(new DefaultUserCreator(processingState, config));
```
Comment (tmetzke) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/user/UserProcessors.java:36]: Good catch, I hadn't thought about that one 🕵️

**Comment by remcowesterhoud** [comment]
@tmetzke I've added the tests, they just return a regular ProblemDetail now 🙂

**Comment by tmetzke** [comment]
> @tmetzke I've added the tests, they just return a regular ProblemDetail now 🙂

Awesome, thanks! I assume our global exception handler catches this now and turns this into a 400. Usually, this would be a [501](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501).

**Comment by remcowesterhoud** [comment]
> > @tmetzke I've added the tests, they just return a regular ProblemDetail now 🙂
> 
> Awesome, thanks! I assume our global exception handler catches this now and turns this into a 400. Usually, this would be a [501](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501).

That's fair. Do you think we should change this now? Should be relatively easy, but then again we should add that to `main` as well and we don't want to forward port this PR I suppose.

**Review by tmetzke — APPROVED** [review]
👍 This let's me sleep better now, thanks! 😄

**Comment by tmetzke** [comment]
> That's fair. Do you think we should change this now? Should be relatively easy, but then again we should add that to `main` as well and we don't want to forward port this PR I suppose.

Nah, good enough in my view.

**Comment by remcowesterhoud** [comment]
OH no merge queue here I see 😅

**Comment by tmetzke** [comment]
YOLO

**Comment by remcowesterhoud** [comment]
If the release branch breaks you know why 👍

**Pull request merged**